modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js

```diff
@@ -309,7 +332,7 @@ class SegmentEdit extends Component {
 
 								<div className="btn-group-item">
 									<ClayButton
-										disabled={this._isQueryEmpty()}
+										disabled={!editing || this._isQueryEmpty()}
```

Comment:
this should be a nicely named constant at the beginning of the render method. Specially because is used twice for pretty much the same thing.